src/ethereum_spec_tools/lint.py

```diff
@@ -205,15 +237,67 @@ def compare(
 
         return diagnostics
 
-    def parse(self, source: str) -> Sequence[str]:
+    def check_import(
+        self, forks: List[Hardfork], position: int, name: str, source: str
+    ) -> List[Diagnostic]:
+        """
+        Checks a Python source and emits diagnostic
+        messages if there are any invalid imports.
+        """
+        diagnostics: List[Diagnostic] = []
+
+        invalid_imports = {
+            "active_fork": forks[position].name,
+            "future_fork": [fork.name for fork in forks[position + 1 :]],
+            "minus2_fork": [fork.name for fork in forks[: position - 1]]
+            if position > 1
+            else [],
+        }
```

Comment:
Couple suggestions here:
 - I don't think the dictionary is necessary?
 - Can directly create the tuples here.
 - Python's array slicing syntax should work without the conditional (but I haven't tried, and could be very wrong.)

```suggestion
    active_fork = forks[position].name
    future_forks = tuple(fork.name for fork in forks[position + 1 :])
    ancient_forks = tuple(fork.name for fork in forks[: position - 1])
```

Reply:
The conditional is required, since the index will be -1 for `frontier` and so, every fork until the latest one will be selected